locust/test/test_main.py

```diff
@@ -1875,11 +1885,30 @@ def test_expect_workers(self):
                 stderr=PIPE,
                 text=True,
             )
-            _, stderr = proc.communicate()
-            self.assertIn("Waiting for workers to be ready, 0 of 2 connected", stderr)
-            self.assertIn("Gave up waiting for workers to connect", stderr)
-            self.assertNotIn("Traceback", stderr)
-            self.assertEqual(1, proc.returncode)
+
+            def check_output():
+                return proc.stderr.readline().strip()
+
+            try:
+                output = ""
+                start_time = time.time()
+                while time.time() - start_time < 2:
+                    line = check_output()
+                    if line:
+                        output += line + "\n"
+                        if "Waiting for workers to be ready" in line or "Gave up waiting for workers" in line:
+                            break
+            finally:
+                proc.terminate()
```

Comment:
the old test verifies that the process shuts down voluntarily, which is the most important feature of `--expect-workers-max-wait`, I think the new test kind of misses that?